internal/provider/features.go

```diff
@@ -20,6 +20,13 @@ func schemaFeatures(supportLegacyTestSuite bool) *pluginsdk.Schema {
 					"purge_soft_delete_on_destroy": {
 						Type:     pluginsdk.TypeBool,
 						Optional: true,
+						Default:  true,
+					},
+
+					"recover_soft_deleted_api_managements": {
```

Comment:
I'm not sure we need to have api management on the end given it is in a block `api_management`? and above we have a property without the post fix applying to the same type of resource?
```suggestion
					"recover_soft_deleted": {
```

Reply:
also i wonder if the purge one should be `purge_soft_deleted_on_destroy` 🤔

Reply:
Thanks, agreed on shortening the first one, the naming didn't quite sit right. For the second one, it's inkeeping with the existing equivalent property for other services so perhaps best to leave it for now.